src/app/InteractionModelEngine.cpp

```diff
@@ -151,6 +168,27 @@ CHIP_ERROR InteractionModelEngine::NewReadClient(ReadClient ** const apReadClien
     return err;
 }
 
+CHIP_ERROR InteractionModelEngine::NewWriteClient(WriteClient ** const apWriteClient, intptr_t aAppIdentifier)
+{
+    CHIP_ERROR err = CHIP_ERROR_NO_MEMORY;
+
+    for (auto & writeClient : mWriteClients)
+    {
+        if (writeClient.IsFree())
+        {
+            *apWriteClient = &writeClient;
+            err            = writeClient.Init(mpExchangeMgr, mpDelegate, aAppIdentifier);
```

Comment:
The "will be set to NULL on error" logic seems to not be held if "NO MEMORY". how about:


```cpp
*apWriteClient = nullptr;

for (auto & writeClient : mWriteClients) {
  if (!writeClient.IsFree()) {
    continue;
  }
  
  ReturnErrorOnFailure(writeClient.Init(....));
  *apWriteClient = &writeClient;
  return CHIP_NO_ERROR;
}

return CHIP_ERROR_NO_MEMORY;
```

Reply:
Done, thanks